benchmarks/hash_table/static_multimap/count_bench.cu

```diff
@@ -14,102 +14,103 @@
  * limitations under the License.
  */
 
+#include <defaults.hpp>
 #include <key_generator.hpp>
 
 #include <cuco/static_multimap.cuh>
 
 #include <nvbench/nvbench.cuh>
 
 #include <thrust/device_vector.h>
+#include <thrust/transform.h>
+
+namespace cuco {
+namespace benchmark {
+
+using namespace defaults;
 
 /**
- * @brief A benchmark evaluating multi-value `count` performance:
- * - Total number of insertions: 100'000'000
- * - CG size: 8
+ * @brief A benchmark evaluating multi-value `count` performance
  */
-template <typename Key, typename Value, dist_type Dist, nvbench::int32_t Multiplicity>
-std::enable_if_t<(sizeof(Key) == sizeof(Value)), void> nvbench_static_multimap_count(
-  nvbench::state& state,
-  nvbench::type_list<Key, Value, nvbench::enum_type<Dist>, nvbench::enum_type<Multiplicity>>)
+template <typename Key, typename Value, typename Dist>
+std::enable_if_t<(sizeof(Key) == sizeof(Value)), void> static_multimap_count(
+  nvbench::state& state, nvbench::type_list<Key, Value, Dist>)
 {
-  auto const num_keys      = state.get_int64("NumInputs");
-  auto const occupancy     = state.get_float64("Occupancy");
-  auto const matching_rate = state.get_float64("MatchingRate");
+  using pair_type = cuco::pair_type<Key, Value>;
 
-  std::size_t const size = num_keys / occupancy;
+  auto const num_keys      = state.get_int64_or_default("NumInputs", N);
+  auto const occupancy     = state.get_float64_or_default("Occupancy", OCCUPANCY);
+  auto const matching_rate = state.get_float64_or_default("MatchingRate", MATCHING_RATE);
 
-  std::vector<Key> h_keys(num_keys);
-  std::vector<cuco::pair_type<Key, Value>> h_pairs(num_keys);
+  std::size_t const size = num_keys / occupancy;
 
-  generate_keys<Dist, Multiplicity, Key>(h_keys.begin(), h_keys.end());
+  thrust::device_vector<Key> keys(num_keys);
 
-  for (auto i = 0; i < num_keys; ++i) {
-    Key key           = h_keys[i];
-    Value val         = h_keys[i];
-    h_pairs[i].first  = key;
-    h_pairs[i].second = val;
-  }
+  key_generator gen;
+  gen.generate<Dist>(state, keys.begin(), keys.end());
 
-  generate_probe_keys<Key>(matching_rate, h_keys.begin(), h_keys.end());
+  thrust::device_vector<pair_type> pairs(num_keys);
+  thrust::transform(
+    thrust::device, keys.begin(), keys.end(), pairs.begin(), [] __device__(Key const& key) {
+      return pair_type(key, {});
+    });
 
-  thrust::device_vector<Key> d_keys(h_keys);
-  thrust::device_vector<cuco::pair_type<Key, Value>> d_pairs(h_pairs);
+  gen.dropout(keys.begin(), keys.end(), matching_rate);
 
-  state.add_element_count(num_keys, "NumKeys");
+  state.add_element_count(num_keys);
+  state.set_global_memory_rw_bytes(num_keys * sizeof(pair_type));
```

Comment:
Should this be `num_keys * sizeof(Key)` instead, since `count` just takes keys as input but doesn't produce any output besides a single integer? @PointKernel any thoughts on this?

Reply:
If we want to track the number of bytes "counted" per second. Then yes, it would be `num_keys * sizeof(key_type)`.